code/game/atom/atom_render.dm

```diff
@@ -0,0 +1,249 @@
+ * Provides sourcing, manages "*" type draws, and allows lookup along the entire "stack" of these effecrs
+ */
+/datum/render_relay
+	///bool if this relay should be drawing the original(provider) as well
+	var/draw_original
+	/**
+	 * This is the target we are relaying to, it can be either an atom or a /datum/filter_data (hence no type)
+	 * Atom and Ma is annoying but easy since we can just set the var
+	 * Because lummox loves us so much var access on filters access it's args instead
+	 * thus Filters will runtime if it's not the right type
+	 * Which means the vars available to access are inconsistent at runtime. yay.
+	 * even worse type doesnt actually check the type it checks the string arg
+	 * instead we have to thus use /datum/filter_data
+	 */
+	var/datum/target
+	///atom that we are rendering from that has a render_target set. CAN be an atom OR an image.
+	var/atom/provider
+
+/datum/render_relay/New(draw_original, datum/target, atom/provider)
+	. = ..()
+	src.draw_original = draw_original
+	src.target = target
+	src.provider = provider
+	LAZYADD(provider.rendering_to, src)
+	if(isatom(target) || isimage(target))
+		var/atom/atom_cast = target
+		atom_cast.rendering_from = src
+		atom_cast.render_source = provider.render_target
+	if(isfilterdata(target))
+		var/datum/filter_data/filt_target = target
+		filt_target.set_render_from(src)
+	RegisterSignal(provider, COMSIG_QDELETING, PROC_REF(qdel_self))
+	RegisterSignal(target, COMSIG_QDELETING, PROC_REF(qdel_self))
+
+///called on del of either provider to clean us up
+/datum/render_relay/proc/qdel_self()
+	SIGNAL_HANDLER
+	provider.stop_render_to(target)
+	qdel(src)
+
+/datum/render_relay/Destroy(force, ...)
+	target = null
+	provider = null
+	return ..()
+
+/**
+ * Links two atoms using render_target and render_source to draw src instead of source
+ * @param target		/datum/filter_data or /atom or /image to draw to
+ * @param draw_original	Bool, whether we want to continue rendering the original atoms appearance
+ */
+/datum/proc/relay_render_to(datum/target, draw_original = TRUE)
+	ASSERT(isatom(src) || isimage(src))
+	ASSERT(isatom(target) || isfilterdata(target) || isimage(target), "Invalid target for rendering relay")
+	if(target in get_below_renderers())
+		CRASH("Circular rendering reference detected. Canceling applying render relay of [target.type]")
+	if(!isfilterdata(target))
+		var/atom/atom_cast = target
+		if(atom_cast.render_source)
+			CRASH("Tried to relay twice to the same atom. Consider using a holder in viscontents instead.")
+	//filterdata sanity checks are handled by add filter already
+
+	//we can be an image or an atom but they have the same vars so hence this cursed shit
+	var/atom/atom_cast = src
+	if(!atom_cast.render_target)
+		var/new_target = ""
+		if(!draw_original)
+			new_target = "*"
+
+		var/static/uuid = 0
+		uuid++
+		new_target = "[new_target]wowza[uuid]wowza"
```

Comment:
```suggestion
		//soooooo lemon thinks that there may be some byond bugs with
		//non-"normal" chars at the end and slates
		//cant be assed to test so idk if its true
		//so lets just add a buffer on both sides just in case(tm)
		new_target = "[new_target]wowza[uuid]wowza"
```